src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java

```diff
@@ -0,0 +1,852 @@
+     */
+    default List<User> getUsers()
+    {
+        return this.getUserCache().asList();
+    }
+
+    /**
+     * This returns the {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel} which has the same id as the one provided.
+     * <br>If there is no known {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel} with an id that matches the provided
+     * one, then this returns {@code null}.
+     *
+     * @param  id The id of the {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel}.
+     *
+     * @return Possibly-null {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel} with matching id.
+     */
+    default VoiceChannel getVoiceChannelById(final long id)
+    {
+        return this.getVoiceChannelCache().getElementById(id);
+    }
+
+    /**
+     * This returns the {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel} which has the same id as the one provided.
+     * <br>If there is no known {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel} with an id that matches the provided
+     * one, then this returns {@code null}.
+     *
+     * @param  id The id of the {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel}.
+     *
+     * @return Possibly-null {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel} with matching id.
+     */
+    default VoiceChannel getVoiceChannelById(final String id)
+    {
+        return this.getVoiceChannelCache().getElementById(id);
+    }
+
+    /**
+     * {@link net.dv8tion.jda.core.utils.cache.SnowflakeCacheView SnowflakeCacheView} of
+     * all cached {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannels} visible to this ShardManager session.
+     *
+     * @return {@link net.dv8tion.jda.core.utils.cache.SnowflakeCacheView SnowflakeCacheView}
+     */
+    default SnowflakeCacheView<VoiceChannel> getVoiceChannelCache()
+    {
+        return SnowflakeCacheView.all(() -> this.getShardCache().stream().map(JDA::getVoiceChannelCache));
+    }
+
+    /**
+     * An unmodifiable list of all {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannels} of all connected
+     * {@link net.dv8tion.jda.core.entities.Guild Guilds}.
+     *
+     * @return Possible-empty list of all known {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannels}.
+     */
+    default List<VoiceChannel> getVoiceChannels()
+    {
+        return this.getVoiceChannelCache().asList();
+    }
+
+    /**
+     * Removes all provided listeners from the event-listeners and no longer uses them to handle events.
+     *
+     * @param  listeners
+     *         The listener(s) to be removed.
+     */
+    default void removeEventListener(final Object... listeners)
+    {
+        this.getShardCache().forEach(jda -> jda.removeEventListener(listeners));
+    }
+
+    /**
+     * Restarts all shards
+     */
+    void restart();
```

Comment:
We should give more details on how this is done, via the SessionReconnectQueue and that it will take shardTotal * 5200 ms